devicetypes/sky-nie/min-smart-plug-dimmer.src/min-smart-plug-dimmer.groovy

```diff
@@ -63,39 +79,128 @@
+        fingerprint mfr: "0312", prod: "AC01", model: "4002", deviceJoinName: "New One Smart Plug Dimmer",  ocfDeviceType: "oic.d.smartplug" //N4002
+        fingerprint mfr: "0312", prod: "0004", model: "EE02", deviceJoinName: "Minoston Dimmer Switch", ocfDeviceType: "oic.d.switch"    //MS11ZS Minoston Smart Dimmer Switch
+        fingerprint mfr: "0312", prod: "EE00", model: "EE04", deviceJoinName: "Minoston Dimmer Switch", ocfDeviceType: "oic.d.switch"    //MS13ZS Minoston Smart Toggle Dimmer Switch
+        fingerprint mfr: "0312", prod: "BB00", model: "BB02", deviceJoinName: "Evalogik Dimmer Switch", ocfDeviceType: "oic.d.switch"    //ZW31S Evalogik Smart Dimmer Switch
+        fingerprint mfr: "0312", prod: "BB00", model: "BB04", deviceJoinName: "Evalogik Dimmer Switch", ocfDeviceType: "oic.d.switch"    //ZW31TS Evalogik Smart Toggle Dimmer Switch
     }
 
     preferences {
-        configParams.each {
-            if (it.range) {
-                input "configParam${it.num}", "number", title: "${it.name}:", required: false, defaultValue: "${it.value}", range: it.range
-            } else {
-                input "configParam${it.num}", "enum", title: "${it.name}:", required: false, defaultValue: "${it.value}", options: it.options
-            }
+        getConfigParamInput(ledModeParam)
+        getConfigParamInput(autoOffIntervalParam)
+        getConfigParamInput(autoOnIntervalParam)
+        getConfigParamInput(powerFailureRecoveryParam)
+        getConfigParamInput(pushDimmingDurationParam)
+        getConfigParamInput(holdDimmingDurationParam)
+        getConfigParamInput(minimumBrightnessParam)
+        input "disclaimer", "paragraph",
+                title: "WARNING",
+                description: "Configuring for 'Night Light Settings' is only valid for the devices with product number of MP21ZD、MP22ZD、N4002(one of them)",
+                required: false
+        getConfigParamInput(nightLightParam)
+        input "disclaimer", "paragraph",
+                title: "WARNING",
+                description: "Configuring for 'Maximum Brightness' and 'Paddle Control' are only valid for the devices with product number of MS11ZS、MS13ZS、ZW31S、ZW31TS(one of them)",
+                required: false
+        getConfigParamInput(maximumBrightnessParam)
+        getConfigParamInput(paddleControlParam)
+    }
+}
+
+private getConfigParamInput(param) {
+    if (param.range) {
+        input "configParam${param.num}", "number", title: "${param.name}:", required: false, defaultValue: "${param.value}", range: param.range
+    } else {
+        input "configParam${param.num}", "enum", title: "${param.name}:", required: false, defaultValue: "${param.value}", options: param.options
+    }
+}
+
+private addChildButton() {
+    log.warn "Creating Button Device"
+    def child = addChildDevice(
+            "smartthings",
```

Comment:
@sky-nie Did You double-check (with a real device) that this child device ("smartthings" namespace) is working correctly with Your DTH ("sky-nie" namespace) ?

Reply:
Yes, I have checked a moment ago, when I set the PaddleControlOptions as "Toggle", The child button can receive "pushed", And I can set the action for child button.

Reply:
But our project manager said that only "pushed" action associations can be realized, obviously there are not as many as expected (the original planned action associations are "pushed", "down", "down_2x", "up", "up_2x" "), since Automation has been enhanced in the SmarthThings App, the functions related to CentralScene can be deleted.

Reply:
But our project manager said that only "pushed" action associations can be realized, obviously there are not as many as expected (the original planned action associations are "pushed", "down", "down_2x", "up", "up_2x" "), since Automation has been enhanced in the SmarthThings App, the functions related to CentralScene can be deleted.